torch/_inductor/lowering.py

```diff
@@ -3721,6 +3722,48 @@ def _realize(x):
     return clone(x)
 
 
+@register_lowering(c10d.traceable_allreduce)
+def allreduce(inputs):
+    return TensorBox.create(
+        ir.AllReduce.create(
+            inputs,
+        )
+    )
```

Comment:
Random thoughts: Have you considered to make it pointwise so that we are able to fuse the communication ops into the loop as well?